e2e/basic/suite_setup_test.go

```diff
@@ -2,41 +2,121 @@ package basic
 
 import (
 	"context"
+	"sync"
+	"sync/atomic"
 	"testing"
+	"time"
 
 	"github.com/sirupsen/logrus"
 	"github.com/stretchr/testify/suite"
+	"k8s.io/apimachinery/pkg/api/resource"
 
+	"github.com/celestiaorg/knuu/e2e"
+	"github.com/celestiaorg/knuu/pkg/instance"
+	"github.com/celestiaorg/knuu/pkg/k8s"
 	"github.com/celestiaorg/knuu/pkg/knuu"
+	"github.com/celestiaorg/knuu/pkg/minio"
 )
 
-type TestSuite struct {
+const (
+	testTimeout = time.Minute * 15 // the same time that is used in the ci/cd pipeline
+
+	nginxImage       = "docker.io/nginx:latest"
+	nginxVolumeOwner = 0
+	nginxPort        = 80
+	nginxHTMLPath    = "/usr/share/nginx/html"
+	nginxCommand     = "nginx -g daemon off"
+
+	resourcesHTML = "../system/resources/html"
+
+	alpineImage = "alpine:latest"
+)
+
+type Suite struct {
 	suite.Suite
-	Knuu *knuu.Knuu
+	Knuu     *knuu.Knuu
+	Executor e2e.Executor
+
+	cleanupMu     sync.Mutex
+	totalTests    atomic.Int32
+	finishedTests int32
```

Comment:
I left this comment on another PR too, but can't we refactor this to deduplicate? 

Can be a follow up.

Reply:
yes thank you. the followup issue is created.